handlers/update-supporter-plus-amount/test/updateSupporterPlusAmountEdgeCases.test.ts

```diff
@@ -0,0 +1,154 @@
+import dayjs from 'dayjs';
+import { buildUpdateAmountRequestBody } from '../src/zuoraApi';
+
+describe('Supporter Plus Amount Update - Edge Cases', () => {
+	describe('Auto-renewed subscription scenarios', () => {
+		test('Should handle subscription with charge effective date before current term start (the original bug)', () => {
+			// Test data representing the problematic scenario:
+			// - Original subscription: 2023-10-26
+			// - Auto-renewed: 2024-04-10 (current term start)
+			// - Charge effective date: 2023-10-26 (original)
+			// - Current term end: 2025-04-10
```

Comment:
we can make all these consts instead of the comments and use them below to reduce duplication and improve clarity and better still make the relative dates clearer
```suggestion
const originalSubscriptionDate = dayjs('2023-10-26');
const termStartDate = originalSubscriptionDate.add(5, 'month');
const chargeEffectiveDate = originalSubscriptionDate;
const currentTermEndDate = originalSubscriptionDate.add(18, 'month');
```

Then in the subscriptiondata you can use `zuoraDateFormat(originalSubscriptionDate)` accordingly